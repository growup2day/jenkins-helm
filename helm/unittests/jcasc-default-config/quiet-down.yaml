templates:
  - config-jcasc.yaml
tests:
  - it: should include quiet down groovy script in default config scripts when enabled
    set:
      JCasC:
        defaultConfig: true
        quietDown:
          enableScript: true
          quietDown: false
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-quiet-down.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-quiet-down.yaml]
  - it: should include correct quietDown value
    set:
      JCasC:
        defaultConfig: true
        quietDown:
          enableScript: true
          quietDown: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-quiet-down.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-quiet-down.yaml]
  - it: should apply the customised quiet down message if specified
    set:
      JCasC:
        defaultConfig: true
        quietDown:
          enableScript: true
          quietDown: true
          quietDownMessage: Custom quiet down message.
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-quiet-down.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-quiet-down.yaml]
  - it: should not include quiet down groovy script in default config scripts when disabled
    set:
      JCasC:
        defaultConfig: true
        quietDown:
          enableScript: false
    asserts:
      - isNull:
          path: data.[jcasc-default-config-quiet-down.yaml]
