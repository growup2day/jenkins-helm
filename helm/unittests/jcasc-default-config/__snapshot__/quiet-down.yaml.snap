should apply the customised quiet down message if specified:
  1: |
    |-
      groovy:
        - script: |
            def quietDown = true
            def jenkins = jenkins.model.Jenkins.get()
            def quietDownMessage = "Custom quiet down message."
            if (quietDown) {
              if (!jenkins.isQuietingDown() ||
                  jenkins.getQuietDownReason() != quietDownMessage) {
                // puts Jenkins into quietdown/shutdown mode, without waiting for existing builds to complete (block=false)
                jenkins.doQuietDown(false, 0, quietDownMessage)
              }
            } else {
              if (jenkins.isQuietingDown()) {
                jenkins.doCancelQuietDown()
              }
            }
should include correct quietDown value:
  1: |
    |-
      groovy:
        - script: |
            def quietDown = true
            def jenkins = jenkins.model.Jenkins.get()
            def quietDownMessage = "This is currently the backup Jenkins instance, use the instance in the other cluster instead."
            if (quietDown) {
              if (!jenkins.isQuietingDown() ||
                  jenkins.getQuietDownReason() != quietDownMessage) {
                // puts Jenkins into quietdown/shutdown mode, without waiting for existing builds to complete (block=false)
                jenkins.doQuietDown(false, 0, quietDownMessage)
              }
            } else {
              if (jenkins.isQuietingDown()) {
                jenkins.doCancelQuietDown()
              }
            }
should include quiet down groovy script in default config scripts when enabled:
  1: |
    |-
      groovy:
        - script: |
            def quietDown = false
            def jenkins = jenkins.model.Jenkins.get()
            def quietDownMessage = "This is currently the backup Jenkins instance, use the instance in the other cluster instead."
            if (quietDown) {
              if (!jenkins.isQuietingDown() ||
                  jenkins.getQuietDownReason() != quietDownMessage) {
                // puts Jenkins into quietdown/shutdown mode, without waiting for existing builds to complete (block=false)
                jenkins.doQuietDown(false, 0, quietDownMessage)
              }
            } else {
              if (jenkins.isQuietingDown()) {
                jenkins.doCancelQuietDown()
              }
            }
